lib/traject/macros/iiif.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module Macros
+  # Macros for extracting Stanford Specific MODS values from Nokogiri documents
+  module IIIF
+    def grab_sul_iiif_links(manifest)
+      resp = Faraday.get URI.parse(manifest)
+      ::JSON.parse(resp.body)
+    rescue Faraday::Error
+      nil
+    end
+
+    def process_iiif_thumbnail(manifest_json)
+      manifest_json['thumbnail']['@id']
+    rescue ::JSON::ParserError
```

Comment:
it was originally added to handle key doesn't exist errors. will see how to better handle and add.